src/main/scala/rocketchip/Periphery.scala

```diff
@@ -173,7 +175,8 @@ trait PeripheryMasterMemModule extends HasPeripheryParameters {
 
   // Abuse the fact that zip takes the shorter of the two lists
   ((io.mem_axi zip coreplexIO.master.mem) zipWithIndex) foreach { case ((axi, mem), idx) =>
-    val axi_sync = PeripheryUtils.convertTLtoAXI(mem)
+    val mem_edge = TileLinkWidthAdapter(mem, edgeMemParams)
```

Comment:
Why don't you factor mem_edge out of the specific axi, ahb, tl connection routines?